run_tests.sh

```diff
@@ -10,77 +19,58 @@ set -ex
 # 5. ineffassign   (https://github.com/gordonklaus/ineffassign)
 # 6. race detector (http://blog.golang.org/race-detector)
 
-# gometalinter (github.com/alecthomas/gometalinter) is used to run each each
+# golangci-lint (github.com/golangci/golangci-lint) is used to run each each
 # static checker.
 
-GOVERSION=${1:-1.11}
+# Default GOVERSION
+[[ ! "$GOVERSION" ]] && GOVERSION=1.11
 REPO=dcrdata
-DOCKER_IMAGE_TAG=dcrdata-golang-builder-$GOVERSION
 
 testrepo () {
   TMPFILE=$(mktemp)
   export GO111MODULE=on
-  
-  golangci-lint run --deadline=10m --disable-all --enable govet --enable staticcheck \
-    --enable gosimple --enable unconvert --enable ineffassign --enable structcheck\
-    --enable goimports --enable misspell --enable unparam
-  if [ $? != 0 ]; then	
-    echo 'golangci-lint has some complaints'	
-    exit 1	
-  fi
+
+  go version
 
   # Test application install
-  go install . ./cmd/...
-  if [ $? != 0 ]; then
-    echo 'go install failed'
-    exit 1
-  fi
+  go build
+  (cd cmd/rebuilddb && go build)
+  (cd cmd/rebuilddb2 && go build)
+  (cd cmd/scanblocks && go build)
 
   # Check tests
   git clone https://github.com/dcrlabs/bug-free-happiness test-data-repo
   tar xvf test-data-repo/stakedb/test_ticket_pool.bdgr.tar.xz -C ./stakedb
 
   env GORACE='halt_on_error=1' go test -v -race ./...
-  if [ $? != 0 ]; then
-    echo 'go tests failed'
-    exit 1
-  fi
+
+  # check linters
+  golangci-lint run --deadline=10m --disable-all --enable govet --enable staticcheck \
+    --enable gosimple --enable unconvert --enable ineffassign --enable structcheck \
+    --enable goimports --enable misspell
```

Comment:
What was the error with `unparam`?  This PR works on my laptop even when I add it back in here.

Reply:
The unparam version must be newer, and it finds legit unparam candidates.  Will address in a future PR.